src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java

```diff
@@ -143,11 +145,21 @@ public final void nativeFinishLayoutOverride(){
 
 	protected abstract void onLayoutCore(boolean changed, int left, int top, int right, int bottom);
 
+	private boolean _isInOnLayoutCore;
+
 	protected final void onLayout(boolean changed, int left, int top, int right, int bottom)
 	{
 		if(!_unoLayoutOverride)
 		{
+			_isInOnLayoutCore = true;
+			_childrenNeedingForceLayout.clear();
 			onLayoutCore(changed, left, top, right, bottom);
+
+			for (int i = 0 ; i < _childrenNeedingForceLayout.size() ; i++) {
```

Comment:
The whole block should be optimized away if _childrenNeedingForceLayout is empty.